interceptor/opencensus/opencensus_test.go

```diff
@@ -154,7 +158,122 @@ func TestOCInterceptor_endToEnd(t *testing.T) {
+	node1 := &commonpb.Node{
+		Identifier:  &commonpb.ProcessIdentifier{Pid: 9489, HostName: "nodejs-host"},
+		LibraryInfo: &commonpb.LibraryInfo{Language: commonpb.LibraryInfo_NODE_JS},
+	}
+	sL1 := []*tracepb.Span{{TraceId: []byte("abcdefghijklmno")}}
+	if err := traceClient.Send(&agenttracepb.ExportTraceServiceRequest{Node: node1, Spans: sL1}); err != nil {
+		t.Fatalf("Failed to send the proxied message from app1: %v", err)
+	}
+
+	// Step 3) Send a trace message without a node but with spans: these
+	// should be registered as belonging to the initiating node.
+	sLi := []*tracepb.Span{{TraceId: []byte("ABCDEFGHIJKLMNO")}, {TraceId: []byte("1234567890abcde")}}
+	if err := traceClient.Send(&agenttracepb.ExportTraceServiceRequest{Node: nil, Spans: sLi}); err != nil {
+		t.Fatalf("Failed to send the proxied message without a node: %v", err)
+	}
+
+	// Step 4) Send a trace message from a differently proxied node "node2" from app2
+	node2 := &commonpb.Node{
+		Identifier:  &commonpb.ProcessIdentifier{Pid: 7752, HostName: "golang-host"},
+		LibraryInfo: &commonpb.LibraryInfo{Language: commonpb.LibraryInfo_GO_LANG},
+	}
+	sL2 := []*tracepb.Span{{TraceId: []byte("_B_D_F_H_J_L_N_")}}
+	if err := traceClient.Send(&agenttracepb.ExportTraceServiceRequest{Node: node2, Spans: sL2}); err != nil {
+		t.Fatalf("Failed to send the proxied message from app2: %v", err)
+	}
+
+	// Give the process sometime to send data over the wire and perform batching
+	<-time.After(150 * time.Millisecond)
+
+	// Examination time!
+	resultsMapping := make(map[string][]*tracepb.Span)
+
+	spanSink.forEachEntry(func(node *commonpb.Node, spans []*tracepb.Span) {
+		resultsMapping[nodeToKey(node)] = spans
+	})
+
+	// First things first, we expect exactly 3 unique keys
+	// 1. Initiating Node
+	// 2. Node 1
+	// 3. Node 2
+	if g, w := len(resultsMapping), 3; g != w {
+		t.Errorf("Got %d keys in the results map; Wanted exactly %d\n\nResultsMapping: %+v\n", g, w, resultsMapping)
+	}
+
+	// Want span counts
+	wantSpanCounts := map[string]int{
+		nodeToKey(initiatingNode): 2,
+		nodeToKey(node1):          1,
+		nodeToKey(node2):          1,
+	}
+	for key, wantSpanCount := range wantSpanCounts {
+		gotSpanCounts := len(resultsMapping[key])
+		if gotSpanCounts != wantSpanCount {
+			t.Errorf("Key=%q gotSpanCount %d wantSpanCount %d", key, gotSpanCounts, wantSpanCount)
+		}
+	}
+
+	// Now ensure that the exported spans match up exactly with the nodes
+	// (or at least their serialized equivalents match up)
+	wantContents := map[string][]*tracepb.Span{
+		nodeToKey(initiatingNode): sLi,
+		nodeToKey(node1):          sL1,
+		nodeToKey(node2):          sL2,
+	}
+
+	gotBlob, _ := json.Marshal(resultsMapping)
+	wantBlob, _ := json.Marshal(wantContents)
+	if !bytes.Equal(gotBlob, wantBlob) {
+		t.Errorf("Unequal serialization results\nGot:\n\t%s\nWant:\n\t%s\n", gotBlob, wantBlob)
+	}
+}
+
+func nodeToKey(n *commonpb.Node) string {
```

Comment:
This is fine for now. Like I mentioned in https://github.com/census-instrumentation/opencensus-service/issues/46#issuecomment-426077190, please add a note that later `Node` could be turned into some other messages that represent monitored resource.

Reply:
This is test code though, just to ensure that the proper nodes were sent over the wire